tar/ccache/ccache_read.c

```diff
@@ -65,7 +68,17 @@ read_rec(void * cookie)
 	ccr->ino = (ino_t)le64dec(ccre.ino);
 	ccr->size = (off_t)le64dec(ccre.size);
 	ccr->mtime = (time_t)le64dec(ccre.mtime);
+#if SIZE_MAX < UINT64_MAX
+	ccr_nch = le64dec(ccre.nch);
```

Comment:
Rather than checking this here, I'd just adjust the check under `/* Sanity check some fields. */` to use the original 64-bit value instead of the `size_t` version.